test/Test/Vector.hs

```diff
@@ -48,25 +74,44 @@ genVector genElem =
 genOrdering :: Monad m => GenT m Ordering
 genOrdering = HG.element [ LT, EQ, GT ]
 
+genVectorIndex :: Monad m => NatRepr n -> GenT m (VectorIndex (n + 1))
+genVectorIndex n =
+  do i <- HG.int (linear 0 (fromIntegral (intValue n - 1)))
+     pure $ elemAtUnsafe i (indicesUpTo n)
 
 instance Show (a -> b) where
   show _ = "unshowable"
 
+orderToOrder :: [Ordering -> Ordering]
```

Comment:
> Maybe change this name to `arbitraryDifferentOrdering` or something

I'm not diametrically opposed to this name, but it sound more to me like the name of a *single* function with type `Ordering -> Ordering`, whereas this is a *list* of such functions. Moreover, one of the functions is `id`, which doesn't give you a "different" ordering, per se. Some other possible names: `orderingEndomorphisms`, `orderingToOrderingFunctions`, `orderingFunctions`.

> and indicate that its main purpose is to choose a function for composition tests.

It's also used in test `imap-const`, I'll add a comment that mentions both of them.

Reply:
I think my core concern is that I spent some time trying to figure out what this function did and what it was for when I first encountered it, and I went about it by reading the function rather than finding the call sites.  The transformations it contained didn't make a lot of sense in the abstract, so it was only when I looked at how it was being used was it clear what it was used for.  I think that the method of capturing that information (in a comment or in the name) is less important than having that information captured somehow, so I'm happy to leave the methodology up to you.

Reply:
Ok, thanks! This has a comment now.